csrc/flash_attn/src/flash_fwd_kernel.h

```diff
@@ -308,6 +310,12 @@ inline __device__ void compute_attn_1rowblock(const Params &params, const int bi
     unsigned long long seed = std::get<0>(seeds);
     unsigned long long offset = std::get<1>(seeds) + (bidb * params.h + bidh) * 32 + tidx % 32;
 
+    // Save seed and offset for backward.
+    if (bidx == 0 && tidx == 0) {
```

Comment:
The grid has 3 dimensions (seqlenq_chunk, batch, head) so we'll need to check that all 3 are zero. Here`bidx==0` only guarantee that batch == 0 and head == 0.

Reply:
Made the change @tridao